### DISCUSSION TITLE
Should import be capable of parsing a JSON string?

### DISCUSSION THREAD

**Discussion by MV10** [webpack/webpack#12478] [Q&A] [answered]
I may be misunderstanding things, but based on [this]() article (the "Using Webpack" section), and other sources like [this]() answer on StackOverflow, it sounds as if `import` should be capable of parsing a JSON string into an object -- but when I try this (via an `externals` entry) all I get back is the string, not an object.

JSON file:
```json
{ "api": "https://localhost:5050/api" }
```
Adding to the externals array:
```javascript
config.externals.push({
  'AppConfig': JSON.stringify(require(path.resolve(configPath, "app-config.json")))
});
```
Usage in a module:
```javascript
import Config from 'AppConfig'; 
```
Using a React component to read the OpenAPI schema:
```javascript
<ApiService schema={Config.api}>
  ...etc
</ApiService>
```
The error message shows the JSON string isn't parsed, it's a string literal:
```
GET https://localhost:9000/%7B%22api%22:%22https://localhost:5050/api%22%7D net::ERR_ABORTED 404
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Naturally, it works if I import the file via path (`import Config from '../../../app-config.json'`) but all that parent-path stuff is what I'm trying to eliminate (quite a few React apps need to share this same config file, it's the basic URIs to our services, auth endpoints, etc.), and we want the literals baked into each module during the build.

Should this work? Am I perhaps missing a plugin or maybe something in Babel? Pretty new to all this JS stuff...

**Comment by alexander-akait** [comment, marked as answer]
Do you need remote file or not? If not you can use https://webpack.js.org/configuration/resolve/#resolvealias or path to `json` file as you described above, if you need remote, you can use `DefinePlugin` and `await import(/* webpackIgnore: true */ DEFINED_PATH_TO_FILE)` https://webpack.js.org/api/module-methods/#magic-comments. `externals` can be used too if you have it on `window` variable

**Comment by MV10** [comment]
No, I don't want to use the file except during the build. The automated CI builds will temporarily modify a local copy of the file with configuration variables pulled from a config management system (Celestial).

I'll look into `resolve` and `alias` -- I tried generating the pathname using `path.resolve` and `path.relative` and putting that into the `exports` array, but for some reason that caused webpack to make runtime calls to retrieve the file from the server as a module -- which doesn't happen if I explicitly define the relative path in the `import` statement. 

Hopefully `alias` will do the trick! Thanks.

**Comment by MV10** [comment]
Setting an `alias` works great, thanks @alexander-akait !
